qiskit/dagcircuit/collect_blocks.py

```diff
@@ -235,6 +292,33 @@ def run(self, block):
         return blocks
 
 
+def split_block_into_layers(block):
+    """Splits a block of nodes into sub-blocks of non-overlapping instructions
+    (or, in other words, into depth-1 sub-blocks).
+    """
+    bit_depths = {}
+    layers = []
+
+    for node in block:
+        cur_bits = set()
+        cur_bits.update(node.qargs)
```

Comment:
```suggestion
        cur_bits = set(node.qargs)
```

Reply:
Changed in [6af8e6f](https://github.com/Qiskit/qiskit-terra/pull/10254/commits/6af8e6fa18eaabb08d742fbb1741e49453bcbe44).